src/wallet/coinselection.cpp

```diff
@@ -93,7 +93,7 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
         bool backtrack = false;
         if (curr_value + curr_available_value < selection_target || // Cannot possibly reach target with the amount remaining in the curr_available_value.
             curr_value > selection_target + cost_of_change || // Selected value is out of range, go back and try other branch
-            (curr_waste > best_waste && (utxo_pool.at(0).fee - utxo_pool.at(0).long_term_fee) > 0)) { // Don't select things which we know will be more wasteful if the waste is increasing
+            (curr_waste > best_waste && (utxo_pool.at(0).fee > utxo_pool.at(0).long_term_fee))) { // Don't select things which we know will be more wasteful if the waste is increasing
```

Comment:
ACK 1a2d94ac586e2ecb55994205fceec14322a949fd

Since the feerate is loop invariant, that boolean will always evaluate the same across one coin selection attempt, and it would be even better to just pull it out of the loop:

```suggestion
    [before loop]
    bool is_high_feerate = utxo_pool.at(0).fee > utxo_pool.at(0).long_term_fee;
    […]
            (is_high_feerate && curr_waste > best_waste)) { // Don't select things which we know will be more wasteful if the waste is increasing
```

Reply:
I considered this as well.  To my surprise moving the comparison outside of the loop didn't make any benchmark difference.  My guess is the compiler must be smart enough to do some optimization, or the benchmark tests aren't sensitive enough to detect a change.

That bieng said, I do think it's more readable outside of the for loop (and doesn't need to be compared every iteration).  @Xekyo does this look better https://github.com/bitcoin/bitcoin/pull/25932/commits/027d59bb4126e350456d0729d8082b13aabc43e9?

Reply:
Maybe this is better as two separate commits.  https://github.com/bitcoin/bitcoin/pull/25932/commits/365aca40453995163bbd17231251512f9f9a103b and https://github.com/bitcoin/bitcoin/pull/25932/commits/953ce061d5e8392d9f4c5bcedd27a2eeb08296db